### ISSUE TITLE
GDBus.Error:org.freedesktop.portal.Error.NotFound when  launching app from "Your first Flutter app" codelab

### ISSUE THREAD

**Issue by Dunrar** [flutter/flutter#139483]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Set up new Flutter **Application** using the **Flutter: New Project** command in VSCode
2. Copy & Paste the initial app from https://codelabs.developers.google.com/codelabs/flutter-codelab-first#2
3. Open `lib/main.dart` and select **Linux (linux-x64)** as build target
4. Press the "play"/"Start Debugging" button

### Expected results

The application starts without errors being thrown.

### Actual results

The Debug Console throws the following error:
```
Launching lib/main.dart on Linux in debug mode...

(test_app:113070): Gdk-CRITICAL **: 16:08:26.947: gdk_window_get_state: assertion 'GDK_IS_WINDOW (window)' failed

** (test_app:113070): CRITICAL **: 16:08:27.243: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: Requested setting not found

** (test_app:113070): CRITICAL **: 16:08:27.244: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: Requested setting not found

** (test_app:113070): CRITICAL **: 16:08:27.245: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: Requested setting not found

** (test_app:113070): CRITICAL **: 16:08:27.245: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: Requested setting not found

** (test_app:113070): CRITICAL **: 16:08:27.246: Failed to read XDG desktop portal settings: GDBus.Error:org.freedesktop.portal.Error.NotFound: Requested setting not found
```

### Code sample

```dart
import 'package:english_words/english_words.dart';
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return ChangeNotifierProvider(
      create: (context) => MyAppState(),
      child: MaterialApp(
        title: 'Namer App',
        theme: ThemeData(
          useMaterial3: true,
          colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepOrange),
        ),
        home: MyHomePage(),
      ),
    );
  }
}

class MyAppState extends ChangeNotifier {
  var current = WordPair.random();
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    var appState = context.watch<MyAppState>();

    return Scaffold(
      body: Column(
        children: [
          Text('A random idea:'),
          Text(appState.current.asLowerCase),
        ],
      ),
    );
  }
}

```

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.16.1, on Manjaro Linux 6.1.64-1-MANJARO, locale en_US.UTF-8)
    • Flutter version 3.16.1 on channel stable at /opt/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7f20e5d18c (7 days ago), 2023-11-27 09:47:30 -0800
    • Engine revision 22b600f240
    • Dart version 3.2.1
    • DevTools version 2.28.3

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /home/donrar/Android/Sdk
    • Platform android-33-ext5, build-tools 33.0.2
    • Java binary at: /opt/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)
    • All Android licenses accepted.

[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop
    • clang version 16.0.6
    • cmake version 3.27.9
    • ninja version 1.11.1
    • pkg-config version 2.0.3

[✓] Android Studio (version 2022.3)
    • Android Studio at /opt/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Manjaro Linux 6.1.64-1-MANJARO

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.

```

</details>

**Comment by domesticmouse** [comment]
@cbracken who should we hassle about Linux failures with Flutter 3.16.1? Given this is GTK erroring out, I doubt this is my intro codelab's fault.

**Comment by jwinarske** [comment]
Some observations:
1. It looks like the fix for this entry is not in 3.16.1:
```
Gdk-CRITICAL **: 16:08:26.947: gdk_window_get_state: assertion 'GDK_IS_WINDOW (window)' failed
``|
It's not critical, and will away on a hot restart.

2. For the remaining entries I suspect it's due to Manjaro Linux and possibly a lack of a common XDG desktop portal interface.

@Dunrar what errors do you get when running the same example on Ubuntu?  Given this is the tested distro for the GTK embedder.
